_pages/docs/testing-with-xquery.md

```diff
@@ -5,17 +5,233 @@ lead_text: ''
 permalink: /docs/testing-with-xquery/
 ---
 
-# Testing With XQuery
+## Testing XQuery With XQuery
 
-*TODO*
+### Scenario 1 - Test assert all exist with assert-equal for success
+```
+xquery version "1.0-ml";
 
-## Testing XQuery With XQuery
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
 
-*TODO*
+let $count := 2
+let $testSequence := (1,2)
+let $result := test:assert-all-exist($count, $testSequence)
+return test:assert-equal($result/@type/string(), "success")
+```
+### Scenario 2 - Test assert all exist with assert-equal for exception error
+```
+xquery version "1.0-ml";
 
-## Test Server-side JavaScript with XQuery
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $count := 3
+let $testSequence := (1,2)
+let $errorMessage := "Assert All Exist failed"
+let $result := 
+try{
+  test:assert-all-exist($count, $testSequence)
+} catch($err) {
+  $err
+}
+return test:assert-equal($result//error:code/string(), $errorMessage)
+```
+### Scenario 3 - Test assert same values with same sequences for left and right
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+let $leftSequence := ("a","b","c")
+let $rightSequence := ("a","b","c")
+return test:assert-same-values($leftSequence, $rightSequence)
+```
+### Scenario 4 - Test assert same values with different sequences for left and right. This generates an exception that is validated with the actual value to get success status
+```
+ xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace error = "http://marklogic.com/xdmp/error";
+
+let $leftSequence := ("a","d","c")
+let $rightSequence := ("a","b","c")
+let $formattedErrorMessage := "Assert Equal failed (ASSERT-EQUAL-FAILED): a c d a b c"
+return 
+try{
+  test:assert-same-values($leftSequence, $rightSequence)
+} catch($err) {
+  test:assert-equal($err//error:format-string/string(), $formattedErrorMessage)
+}
+```
+### Scenario 5 - Test assert equal xml - all attributes and elements matched
+```
+xquery version "1.0-ml";
+
+import module namespace test="http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+declare namespace t = "http://marklogic.com/roxy/test";
+
+let $actual := <t:test type="actual" xmlns:t="http://marklogic.com/roxy/test"><code>abc</code></t:test>
```

Comment:
let's simplify the example and not use namespaced XML.  Let's use something more descriptive and less nuanced.
```
<element attribute="attribute-value">element value</element>
```